R/compile_report.R

```diff
@@ -1,19 +1,47 @@
-#' Compile an rmarkdown report
+#' Compile an R Markdown report
 #'
 #' This function can be used to compile a report using its name, or a partial
 #' match for its name. The full path needs not be specified, but reports are
-#' expected to be inside the \code{report_sources} folder (or any subfolder
+#' expected to be inside the \code{report_sources} folder (or any sub-folder
 #' within). Outputs will be generated in a named and time-stamped directory
 #' within \code{report_outputs}.
 #'
+#' ## This function will:
+#'
+#' 1. perform cleaning of `report_sources/` if `clean_report_sources = TRUE`;
+#' this will remove pretty much anything that's not an `Rmd` file or a
+#' `README`; this is externalised in the function `clean_report_sources`
+#'
+#' 2. check that input file exists; identify all files in `report_sources/`
+#' prior to the compilation, used later to diff post-compilation to identify
+#' new files
+#'
+#' 3. compile the user-provided report `foo_date[].Rmd` file stored in
+#' `report_sources/`; compilation parameters are passed to
+#' `rmarkdown::render`'s argument `params` through `render_param`; parameters
```

Comment:
It is a minor detail, but actually at some point @amygimma and I bumped into something that looked like a bug with `params` being ignored by `rmarkdown::render`, so the new approach is technically passing  an environment which contains a variable called `params`.

Reply:
i'll edit this and push a commit after i merge the passing branch